### PR TITLE
Show branch names when listing patches

### PR DESCRIPTION
**Pull request by slack-coder**
Show branch names when listing patches, or nothing when none are present.  This gives the following output:

```
Update README hnrkkqa8n56 R0 f092fb4(change-readme,rad/change-readme) ahead 1, behind 0
└── opened by hnrkd471zrxzjgct4gestio35ai7wnhq67yeo 30 minutes ago
```

### CODE REVIEW

File: scripts/contributor-workflow-setup.sh [hunk 1]
```diff
@@ -110,6 +110,7 @@ CONTRIBUTOR_PEER=$(rad self --peer)
 cd acme
 
 # Create change
+git checkout -b update-readme
```
Comment (cloudhead) [inline scripts/contributor-workflow-setup.sh:113]: Does that not break anything?

File: patch/src/lib.rs [hunk 2]
```diff
@@ -555,13 +556,40 @@ fn pretty_sync_status(
...
+        for r in repo.references()?.flatten() {
+            if let (Some(oid), Some(name)) = (&r.target(), &r.shorthand()) {
+                if oid == revision_oid {
+                    branches.push(name.to_string());
+                };
+            };
+        }
+    };
+    if !branches.is_empty() {
+        str = format!("{}({})", str, &branches.join(","));
```
Comment (cloudhead) [inline patch/src/lib.rs:579]:
```suggestion
        str = format!("{} ({})", str, &branches.join(", "));
```

File: patch/src/lib.rs [hunk 3]
```diff
@@ -555,13 +556,40 @@ fn pretty_sync_status(
     Ok(format!("ahead {}, behind {}", ahead, behind))
 }
 
+/// Make a human friendly string for commit version information.
+///
+/// For example 'oid(branch1[,branch2])'.
```
Comment (cloudhead) [inline patch/src/lib.rs:561]:
```suggestion
/// For example '<oid> (branch1, branch2)'.
```

File: patch/src/lib.rs [hunk 4]
```diff
@@ -555,13 +556,40 @@ fn pretty_sync_status(
...
+fn pretty_commit_version(
+    revision_oid: &git::Oid,
+    repo: &Option<git::Repository>,
+) -> anyhow::Result<String> {
+    let mut str = common::fmt::oid(revision_oid);
+
+    let mut branches: Vec<String> = vec![];
```
Comment (cloudhead) [inline patch/src/lib.rs:568]: I'd group the `let` declarations

**Comment by cloudhead** [comment]
Looks good overall, just a few minor things.

File: scripts/contributor-workflow-setup.sh (hunk 1 above)
Comment (slack-coder) [reply scripts/contributor-workflow-setup.sh:113]: Nope.  Yet I updated it to checkout master afterwards to avoid any future problems.

Comment (cloudhead) [reply scripts/contributor-workflow-setup.sh:113]: For me it's broken:
```
fatal: commit hashes do not match: 1d09e3c1591bb81d63a78bc86a1a89cb38c2681d vs. c28ebbf6fe1ef53e4908124369f1a531e3f722f6
```

Comment (cloudhead) [reply scripts/contributor-workflow-setup.sh:113]: What was the reason for changing it?

Comment (slack-coder) [reply scripts/contributor-workflow-setup.sh:113]: My bad, I didn't notice the error at the end.

The rationale is it being the common workflow to use a feature branch.  The catalyst was the change in this PR to show the branch.

Comment (slack-coder) [reply scripts/contributor-workflow-setup.sh]: As of now, it appears problematic to do this, so I've reverted the commits re: the feature branch.

File: patch/src/lib.rs [hunk 5]
```diff
@@ -555,13 +556,40 @@ fn pretty_sync_status(
...
+) -> anyhow::Result<String> {
+    let mut str = common::fmt::oid(revision_oid);
+    let mut branches: Vec<String> = vec![];
+
+    if let Some(repo) = repo {
+        for r in repo.references()?.flatten() {
+            if let (Some(oid), Some(name)) = (&r.target(), &r.shorthand()) {
+                if oid == revision_oid {
+                    branches.push(name.to_string());
+                };
```
Comment (cloudhead) [inline patch/src/lib.rs:574]: Ok one thing I noticed here, is I'm getting things other than branches match, eg. remote refs that aren't associated to local branches. So I think we need an additional check, maybe by trying to fetch the branch or something?

Comment (cloudhead) [reply patch/src/lib.rs:570]: Going to merge this already, can tackle it later.

**Pull request merged**